cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/ToolCommand.java

```diff
@@ -344,181 +437,113 @@ private void pullToolFromCentral(String supportedPlatform, Path balaCacheDirPath
             // Ignore 'Settings.toml' parsing errors and return empty Settings object
             settings = Settings.from();
         }
+        System.setProperty(CentralClientConstants.ENABLE_OUTPUT_STREAM, "true");
         CentralAPIClient client = new CentralAPIClient(RepoUtils.getRemoteRepoURL(),
                 initializeProxy(settings.getProxy()), settings.getProxy().username(),
                 settings.getProxy().password(), getAccessTokenOfCLI(settings));
         String[] toolInfo = client.pullTool(toolId, version, balaCacheDirPath, supportedPlatform,
                 RepoUtils.getBallerinaVersion(), false);
-        org = toolInfo[0];
-        name = toolInfo[1];
-        version = toolInfo[2];
-    }
+        boolean isPulled = Boolean.parseBoolean(toolInfo[0]);
+        org = toolInfo[1];
+        name = toolInfo[2];
+        version = toolInfo[3];
 
-    private void insertToBalToolsTomlFile() {
-        BalToolsToml distSpecificToolsToml = BalToolsToml.from(distSpecificToolsTomlPath);
-        BalToolsManifest distSpecificToolsManifest = BalToolsManifestBuilder.from(distSpecificToolsToml).build();
-        boolean shouldUpdateToml = true;
-
-        // Should not update the toml version if there is already an already installed higher version
-        if (distSpecificToolsManifest.tools().containsKey(toolId)) {
-            SemanticVersion currentVersionInToml = SemanticVersion.from(
-                    distSpecificToolsManifest.tools().get(toolId).version());
-            if (currentVersionInToml.greaterThanOrEqualTo(SemanticVersion.from(version))) {
-                shouldUpdateToml = false;
-            }
+        if (isPulled) {
```

Comment:
Is this check required since it's done before pulling (in L416) also?

Reply:
The first `isToolAvailableLocally(toolId, version)` can successfully check if the tool is already available only if the user has specified the version explicitly. In that case we skip the pulling part and update the `bal-tools.toml`.

However in the case of `bal tool pull <tool-id>` where version is not mentioned, `isToolAvailableLocally` is not capable of checking the tool availability. Only when we attempt the Central endpoint call to pul, we get the version and then we check for the availability.